core/shared/src/main/scala/fs2/internal/InterruptionOutcome.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2013 Functional Streams for Scala
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package fs2.internal
+
+private[fs2] sealed trait InterruptionOutcome
+
+private[fs2] object InterruptionOutcome {
+  case class Completed(token: Token) extends InterruptionOutcome
+  case class Errored(t: Throwable) extends InterruptionOutcome
+  case object Canceled extends InterruptionOutcome
```

Comment:
o_O

Where exactly does this come from?

Reply:
@djspiewak What do you mean?

Reply:
Actually backing up a bit, I was initially concerned about the bare nature of `Completed`, but I realized that CE itself actually uses this shape internally in a few places.

```scala
type InterruptionOutcome = Outcome[Id, Throwable, Token]
```

Could use `Need` instead if `Id` makes you as nervous as it makes me. `PureConc` has something a bit like this internally. Obviously this is an internal type, so it doesn't matter that much either way, but I just wanted to think about it a bit to make sure there wasn't something important missing in CE3 w.r.t. this type of case.

Reply:
Cool, using type alias worked out nicely.